File: CGtk/CGtk/CategoriaDao.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using Serpis.Ad;
namespace CGtk {
    public class CategoriaDao {

        private static string selectAll = "select * from categoria order by id";

            public static IEnumerable GetAll() {
            IList<Categoria> categorias = new List<Categoria> ();

            IDbCommand dbCommand = App.Intance.DbConnection.CreateCommand ();
            dbCommand.CommandText = selectAll;
            IDataReader dataReader = dbCommand.ExecuteReader ();

            while (dataReader.Read ()) {

                //Opción 1

                categorias.Add (new Categoria ((ulong)dataReader["id"], (string)dataReader["nombre"]));

                //Opción 2

                //Categoria categoria = new Categoria ();
                //categoria.Id =(ulong)dataReader["id"];
                //categoria.Nombre = (string)dataReader["nombre"];

                //Opcion 3

                //Type type = typeof (Categoria);
                //Categoria categoria = Activator.CreateInstance<Categoria>();
                //type.GetProperty ("Id").SetValue (categoria, dataReader["id"]);
                //type.GetProperty ("Nombre").SetValue (categoria, dataReader["nombre"]);


                //categorias.Add(categoria);
            }
            dataReader.Close ();
            return categorias;
        }
        private static string select = "update categoria set nombre=@nombre where id=@id";

        public static Categoria Load(object id) {
            Categoria categoria = new Categoria ();
            IDbCommand dbCommand = App.Intance.DbConnection.CreateCommand ();
            dbCommand.CommandText = select;
            DbCommandHelper.AddParameter (dbCommand, "id", id);
            IDataReader dataReader = dbCommand.ExecuteReader ();
            dataReader.Read ();
            categoria.Id = (ulong)id;
            categoria.Nombre = (string)dataReader["nombre"];
            dataReader.Close ();
            return categoria;
        }

        public static void Save(Categoria categoria) {
            if (categoria.Id == 0) {

                insert(categoria);


            } else {

                update (categoria);

            }

        }

        private static string insertSql = "insert into categoria(nombre) values(@nombre)";
        private static void insert(Categoria categoria) {
            IDbCommand dbCommand = App.Intance.DbConnection.CreateCommand ();
            dbCommand.CommandText = insertSql;
            DbCommandHelper.AddParameter (dbCommand, "nombre", categoria.Nombre); 

        }

        private static string updateSql = "update categoria set nombre=@nombre where id=@id";

        private static void update(Categoria categoria) {

            IDbCommand dbCommand = App.Intance.DbConnection.CreateCommand();
            dbCommand.CommandText = updateSql;
            DbCommandHelper.AddParameter (dbCommand, "nombre", categoria.Nombre);
            DbCommandHelper.AddParameter (dbCommand, "id", categoria.Id);
            dbCommand.ExecuteNonQuery();

        }

        private static string deleteSql = "delete from categoria where id=@id";
        public static void delete(Object id) {
            IDbCommand dbCommand = App.Intance.DbConnection.CreateCommand ();
            dbCommand.CommandText = deleteSql;
            DbCommandHelper.AddParameter (dbCommand, "id", id);

        }

    }
 }

File: GCategoria/src/main/java/iesserpis/ad/CategoriaDao.java
package iesserpis.ad;

import java.sql.PreparedStatement;
import java.sql.SQLException;

public class CategoriaDao {
	
	
	
	/**
	 * Carga la categoria con el id indicado de la base de datos o null si no existe.
	 * @param id
	 * @return categoria con ese id o null si no existe
	 */
	
	public static Categoria load(Object id) {
		return null;
	}
	public static int save(Categoria categoria) throws SQLException {
		if(categoria.getId()==0) {
			insert(categoria);
		}else {
			update(categoria);
		}
		return 0;
		
	}
	private static String insertSql="insert into categoria(nombre) values (?)";
	private static int insert(Categoria categoria) throws SQLException {
		
		try(PreparedStatement preparedStatement = App.getInstance().getConnection().prepareStatement(insertSql)){
			preparedStatement.setString(1, categoria.getNombre());
			return preparedStatement.executeUpdate();

		}
		
	}
	private static String updatesql="update categoria set nombre=? where id=?";
	private static int update(Categoria categoria)throws SQLException {
		
		try(PreparedStatement preparedStatement=App.getInstance().getConnection().prepareStatement(updatesql)){
			
			preparedStatement.setString(1, categoria.getNombre());
			preparedStatement.setObject(2, categoria.getId());
		
			return preparedStatement.executeUpdate();
		}
	
	}
	private static String deletesql="delete from categoria where id=?";
	private static int delete(Categoria categoria)throws SQLException {
		
		try(PreparedStatement preparedStatement=App.getInstance().getConnection().prepareStatement(deletesql)){
		
			preparedStatement.setObject(1, categoria.getId());
		
			return preparedStatement.executeUpdate();
		}
	
	}


}
